pkg/k8shandler/visualization.go

```diff
@@ -0,0 +1,49 @@
+package k8shandler
+
+import (
+  "k8s.io/apimachinery/pkg/api/errors"
+  "github.com/sirupsen/logrus"
+  "github.com/openshift/cluster-logging-operator/pkg/utils"
+
+  sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
+  logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
+)
+
+func CreateOrUpdateVisualization(logging *logging.ClusterLogging) error {
+  return createOrUpdateKibanaSecret(logging)
+}
+
+func createOrUpdateKibanaSecret(logging *logging.ClusterLogging) error {
+
+  kibanaSecret := utils.Secret(
+    "logging-kibana",
+    logging.Namespace,
+    map[string][]byte{
+      "ca": utils.GetFileContents("/tmp/_working_dir/ca.crt"),
+      "key": utils.GetFileContents("/tmp/_working_dir/system.logging.kibana.key"),
+      "cert": utils.GetFileContents("/tmp/_working_dir/system.logging.kibana.crt"),
+    }  )
+
+  err := sdk.Create(kibanaSecret)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure constructing Kibana secret: %v", err)
+  }
+
+  proxySecret := utils.Secret(
+    "logging-kibana-proxy",
+    logging.Namespace,
+    map[string][]byte{
+      "oauth-secret": utils.GetRandomWord(64),
+      "session-secret": utils.GetRandomWord(32),
+      "server-key": utils.GetFileContents("/tmp/_working_dir/kibana-internal.key"),
+      "server-cert": utils.GetFileContents("/tmp/_working_dir/kibana-internal.crt"),
+//      "server-tls.json": utils.GetFileContents("/tmp/_working_dir/server-tls.json"),
```

Comment:
@josefkarasek does the oauth-proxy still make use of this file?

Reply:
I don't believe so.  The configuration for the proxy is here and I don't see a similar file. https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_logging_kibana/templates/kibana.j2#L98-L110